File: 3. Array/2Pointers.py
def TwoSum(arr, n, x):

    i = 0
    j = n-1

    while (i < j):

        if (arr[i]+arr[j] == x):
            return 1
        elif (arr[i]+arr[j] < x):
            i += 1
        else:
            j -= 1

    return 0


arr = [2, 3, 5, 8, 9, 10, 11]

val = 17

res = TwoSum(arr, len(arr), val)

if(res==1):
    print("The Sum exists")
else:
    print("Sum does not exist")

File: 3. Array/2Pointers.cpp
// This is the program for finding if the given 2sum exists in an array using 2 Pointer technique
#include <iostream>
#include <stdio.h>
#include <conio.h>

int TwoSum(int arr[], int n, int x)
{
    int i = 0;
    int j = n - 1;

    while (i < j)
    {
        if (arr[i] + arr[j] == x)
        {
            return 1;
        }
        else if (arr[i] + arr[j] < x)
        {
            i++;
        }
        else
        {
            j--;
        }
    }

    return 0;
}

int main()
{
    system("cls");

    int arr[] = {2, 3, 5, 8, 9, 10, 11};
    int n = sizeof(arr) / sizeof(arr[0]);

    int res = TwoSum(arr, n, 11);

    if (res == 1)
    {
        std::cout << "Sum exists";
    }
    else
    {
        std::cout << "Sum does not exist";
    }

    return 0;
}